unix/xserver/hw/vnc/vncInput.c

```diff
@@ -50,7 +50,7 @@ extern const unsigned int code_map_qnum_to_xorgevdev_len;
 extern const unsigned short code_map_qnum_to_xorgkbd[];
 extern const unsigned int code_map_qnum_to_xorgkbd_len;
 
-#define BUTTONS 7
+#define BUTTONS 9
```

Comment:
If we're extending things, we might as well try to support everything.

evdev (which is the reference) has 8 buttons defined currently. Adding the 4 scroll buttons puts it at 12 buttons.

If we look at the Xorg evdev driver, it allocates room for 16 buttons, plus the 4 scroll buttons. Which means that we need more than 16 bits. I'm not sure if there is support for any such mice, though?

Perhaps we should at least use the full 16 (12+4) buttons we've made room for?

Reply:
yeah, if we're extending functionality it would be nice to fully extend it.